circuit_knitting/cutting/cutting_reconstruction.py

```diff
@@ -72,7 +72,7 @@ def reconstruct_expectation_values(
         for label, subobservable in observables.items():
             if any(obs.phase != 0 for obs in subobservable):
                 raise ValueError("An input observable has a phase not equal to 1.")
-        subobservables_by_subsystem = observables
+        subobservables_by_subsystem = observables  # type: ignore
```

Comment:
I believe this is a false positive from mypy. The variable is a `dict[Hashable, Any]`, and the expression is a `dict[str | int, Any]`. I believe the expression lies in a subset of the variable, which should be fine.

Reply:
https://mypy.readthedocs.io/en/stable/generics.html#variance-of-generic-types explains why `dict` is considered an invariant type rather than a covariant type.  But yes, I think ignoring this makes the most sense for us for now.

Reply:
Cool, since this is technically correct, this means we have "correct" type hints with respect to `Hashable` and `str | int`. All the functions correctly indicate what functionality they expect